### ISSUE TITLE
Multiple user.js files left over on Android device

### ISSUE THREAD

**Issue by jonalmeida** [mozilla/web-ext#1919]
### Is this a feature request or a bug?
Bug

### What is the current behavior?

1. Run a web extension using `web-ext` a couple of times.
2. You can find multiple `user.js` files for every run filing up your device.
  - For example: `/storage/emulated/0/web-ext-artifacts-1589424536174/profile/user.js`
  - The Google Files app reports that there are multiple duplicates of these files as well (see screenshot below).

### What is the expected or desired behavior?
1. At most one user.js file with the old ones cleaned up.

### Version information (for bug reports)

* **Firefox version**: GeckoView 79.0, Reference Browser
* **Your OS and version**: Android 10, Pixel 4
* Paste the output of these commands:

````
node --version && npm --version && web-ext --version

v14.4.0
6.14.4
4.2.0
````

<img width="634" alt="Screen Shot 2020-06-03 at 1 45 50 PM" src="https://user-images.githubusercontent.com/1370580/83676372-902a4700-a5a8-11ea-8417-6435b4132e24.png">

**Comment by Rob--W** [comment]
Are you allowing `web-ext` to exit gracefully? What's the output with `-v` to `web-ext run` ?

The `web-ext-artifacts-*` directory should be removed at the end of the test: https://github.com/mozilla/web-ext/blob/ffe57268cd946104b0213f001c712ad2cb2ef78e/src/extension-runners/firefox-android.js#L259-L262
 A known case where it doesn't happen is when the phone is unplugged before that happens. It could also happen if `web-ext` is killed before completion.

**Comment by jonalmeida** [comment]
>  A known case where it doesn't happen is when the phone is unplugged before that happens. It could also happen if `web-ext` is killed before completion.

This is probably the cause and is a common flow as well. Maybe performing a pre-check cleanup on the next run would work as a solution?

My usage of web-ext varies but the next time I'll try to get verbose logs. I was only made aware of these duplicates after they piled up and my phone asked me if I wanted to delete these duplicate files as part of a way to make more space on disk.

**Comment by Rob--W** [comment]
> >  A known case where it doesn't happen is when the phone is unplugged before that happens. It could also happen if `web-ext` is killed before completion.
> 
> This is probably the cause and is a common flow as well. Maybe performing a pre-check cleanup on the next run would work as a solution?

That's #1591 and WIP.

**Comment by rpl** [comment]
Closing as a duplicate of #1591 (which has a fix already merged in the main development branch, will be released as part of web-ext v5.0.0).
